platform/ui/src/components/userPreferencesForm/WindowLevelPreferences.js

```diff
@@ -1,94 +1,105 @@
-import React, { Component } from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
-import './WindowLevelPreferences.styl';
 
-export class WindowLevelPreferences extends Component {
-  constructor(props) {
-    super(props);
+import './WindowLevelPreferences.styl';
 
-    this.state = {
-      data: this.props.windowLevelData,
-    };
-  }
+function WindowLevelPreferencesRow({
+  description,
+  window,
+  level,
+  rowName,
+  onSuccessChanged,
+  onFailureChanged,
+}) {
+  const [rowState, setRowState] = useState({ description, window, level });
 
-  static propTypes = {
-    windowLevelData: PropTypes.object.isRequired,
-    onChange: PropTypes.func,
+  const onInputChanged = (event, name) => {
+    const newValue = event.target.value;
+    setRowState({ ...rowState, [name]: newValue });
   };
 
-  onChange(event, key, field) {
-    const data = this.state.data;
-    const entry = data[key];
-    entry[field] = event.target.value;
-    this.setState({ data });
+  useEffect(() => {
+    onSuccessChanged(rowName, rowState);
```

Comment:
Same here to be aligned with hooks rules. `  useEffect(() => {
    onSuccessChanged(rowName, rowState);
  }, [onSuccessChanged, rowName, rowState]);`

Reply:
Current implementation belongs to component impl only. Behaviourally speaking there is no need to bind changes of  onSuccessChanged, rowName, because if that happens the child component will render anyway